Sources/Target/Linux/Thread.cpp

```diff
@@ -111,7 +108,17 @@ ErrorCode Thread::updateStopInfo(int waitStatus) {
         _stopInfo.reason = StopInfo::kReasonBreakpoint;
         break;
       default:
-        DS2BUG("unknown sigtrap code");
+        if (si.si_code <= 0 && si.si_pid != getpid()) {
+          // If the value of si_code is <= 0, then the signal was generated by a
+          // user process via kill(2), sigsend(2), abort(3C), raise(3C), etc. No
+          // action is needed other than updating the thread's stop info.
+          DS2LOG(Warning, "sigtrap code %d recieved from %" PRI_PID, si.si_code,
+                si.si_pid);
+          _stopInfo.reason = StopInfo::kReasonTrap;
+        } else {
+          DS2BUG("unknown sigtrap code %d", si.si_code);
```

Comment:
If this is a noreturn macro, I think that we could simplify this to:

```c
if (si.si_code > 0 || si.si_pid == getpid())
  DS2BUG(...);
...
```

It would remove the level of indentation and make the condition easier to read.